integration/spark/spark31/build.gradle

```diff
@@ -0,0 +1,157 @@
+import org.apache.tools.ant.filters.ReplaceTokens
+import groovy.io.FileType
+
+import java.nio.file.Files
+
+
+plugins {
+    id 'java'
+    id 'java-library'
+    id 'java-test-fixtures'
+    id 'com.diffplug.spotless' version '6.12.0'
+    id "com.adarshr.test-logger" version "3.2.0"
+    id "org.gradle.test-retry" version "1.5.8"
+    id "com.github.johnrengelman.shadow" version "8.1.1"
+    id "pmd"
+}
+
+pmd {
+    consoleOutput = true
+    toolVersion = "6.46.0"
+    rulesMinimumPriority = 5
+    ruleSetFiles = rootProject.files("pmd-openlineage.xml")
+    ruleSets = []
+    ignoreFailures = true
+}
+
+pmdMain {
+    dependsOn shadowJar
+    reports {
+        html.required = true
+    }
+}
+
+pmdTest {
+    dependsOn shadowJar
+}
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+    maven {
+        url = 'https://astronomer.jfrog.io/artifactory/maven-public-libs-snapshot'
+    }
+}
+
+archivesBaseName='openlineage-spark-spark3'
+
+ext {
+    assertjVersion = '3.25.1'
+    junit5Version = '5.10.1'
+    mockitoVersion = '4.11.0'
+    sparkVersion = '3.1.3'
+    jacksonVersion = '2.15.3'
+    lombokVersion = '1.18.30'
+}
+
+dependencies {
+    compileOnly "org.projectlombok:lombok:${lombokVersion}"
+    annotationProcessor "org.projectlombok:lombok:${lombokVersion}"
+    testCompileOnly "org.projectlombok:lombok:${lombokVersion}"
+    testAnnotationProcessor "org.projectlombok:lombok:${lombokVersion}"
+
+    implementation(project(path: ":shared", configuration: "shadow"))
+    implementation(project(path: ":spark3", configuration: "shadow"))
+
+    compileOnly "org.apache.spark:spark-core_2.12:${sparkVersion}"
+    compileOnly "org.apache.spark:spark-sql_2.12:${sparkVersion}"
+    compileOnly "org.apache.spark:spark-hive_2.12:${sparkVersion}"
+    compileOnly "org.apache.spark:spark-sql-kafka-0-10_2.12:${sparkVersion}"
+    compileOnly "org.apache.iceberg:iceberg-spark3-runtime:0.13.0"
+    compileOnly "io.delta:delta-core_2.12:1.1.0"
```

Comment:
I know we often already got messy for other version's specific Spark modules, but I would give a try for this one to include only dependencies we need. Especially that it is a single class with a test, so this shouldn't be time consuming.

Reply:
I reduced the dependencies, which are not necessary. Also, I would suggest running the `nebula.lint` plugin to find all the dependencies unused because it results in around 300 warnings, probably some of which are false positives but others not.